frontend/src/theme/components/Toggle.ts

```diff
@@ -0,0 +1,109 @@
+import { ComponentMultiStyleConfig } from '@chakra-ui/theme'
+import { getColor } from '@chakra-ui/theme-tools'
+// Call it Switch to be consistent with Chakra UI so the styles
+// are merged correctly
+export const TOGGLE_THEME_KEY = 'Switch'
+
+const parts = [
+  'container', // default container for the toggle itself
+  'track', // oval part along which circle travels
+  'thumb', // circle part
+  'textContainer', // container for label and description
+  'label',
+  'description',
+  'overallContainer', // container for our custom component which contains text + toggle
+  'icon', // icon in centre of thumb
+]
+
+export const Toggle: ComponentMultiStyleConfig = {
+  parts,
+  baseStyle: ({ theme }) => ({
+    overallContainer: {
+      display: 'flex',
+      justifyContent: 'space-between',
+    },
+    label: {
+      textStyle: 'subhead-1',
+      color: 'secondary.700',
+      m: 0,
+    },
+    description: {
+      textStyle: 'body-2',
+      color: 'secondary.400',
+    },
+    container: {
+      // To allow container to have a blue border on focus
+      positive: 'relative',
+      // Spacing between text and track
+      ml: '1rem',
+    },
+    track: {
+      bg: 'neutral.400',
+      _checked: {
+        bg: 'success.500',
+      },
+      p: 0,
+      _focus: {
+        // disable Chakra UI's default shadow
+        boxShadow: `none`,
```

Comment:
Could we not have used a double box shadow?

```
boxShadow: `0 0 0 2px white, 0 0 0 3px var(--chakra-colors-primary-500)`;
```

Reply:
excuse me while I go drown myself